c10/util/TypeSafeSignMath.h

```diff
@@ -79,7 +79,7 @@ template <typename Limit, typename T>
 inline constexpr bool greater_than_max(const T& x) {
   constexpr bool can_overflow =
       std::numeric_limits<T>::digits > std::numeric_limits<Limit>::digits;
-  return can_overflow && x > std::numeric_limits<Limit>::max();
+  return can_overflow && x > (std::numeric_limits<Limit>::max)();
```

Comment:
Can you help me to understand why adding parentheses helps it to compile?

Reply:
> I think spelling out the global namespace would help too, like `::std::numeric_limits<Limit>::max`?

@desertfire I tried it, it doesn't work:
<img width="667" height="371" alt="image" src="https://github.com/user-attachments/assets/9eae3e32-d075-4d51-973b-92080f9eceae" />